packages/syntax-floating-components/src/ariakit-utils/useForwardFocus.ts

```diff
@@ -0,0 +1,69 @@
+import { type PopoverStore } from "@ariakit/react";
+import { getFirstTabbableIn } from "@ariakit/core/utils/focus";
+import {
+  type FocusEventHandler,
+  type FocusEvent,
+  useCallback,
+  useRef,
+  useEffect,
+} from "react";
+
+type OnFocusHandler = FocusEventHandler<HTMLElement>;
+
+export default function useForwardFocus(
+  store: PopoverStore,
+  options?: { onFocus?: OnFocusHandler },
+): OnFocusHandler {
+  const onFocusProp = options?.onFocus;
+  const anchorRef = useRef<HTMLElement | null>(null);
+  const anchorElement = store.useState("anchorElement");
+  // transfer focus to child element if it is focusable
+  //
+  // if there is a focusable child in the trigger children,
+  // focus is immediately transferred to that child element
+  // (e.g. when the user tabs to the trigger)
+  // prevents a double-focus situation where user tabs/steps to the trigger,
+  // then has to step again to focus the child element
+  const forwardFocusToInteractiveChild = useCallback(
+    (evt: FocusEvent<HTMLElement>) => {
+      // only operate on focus events for the tooltip anchor directly, not focus events bubbling up from children
+      if (evt.currentTarget !== evt.target) return;
+      // try to find the first focusable child element if it exists
+      const focusableChild = getFirstTabbableIn(evt.currentTarget, false, true);
+      if (!focusableChild) return;
+      anchorRef.current = evt.currentTarget;
+      store.setAnchorElement(focusableChild);
+      // transfer focus to the child element
+      focusableChild.focus();
+    },
+    [store],
+  );
+
+  useEffect(() => {
+    if (
+      !anchorRef.current ||
+      !anchorElement ||
+      anchorElement === anchorRef.current
+    )
+      return;
+    // ariakit Tooltip uses html data attr to determine if mouse moves closes the tooltip or not
```

Comment:
@benjroy this doesn't feel great... seems like we need to know about Ariakit internals to make this work. Let's definitely add tests around this.